java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -430,7 +430,7 @@ public boolean equals(int index, ValueVector to, int toIndex) {
     if (to == null) {
       return false;
     }
-    if (this.getClass() != to.getClass()) {
+    if (!this.getField().getType().equals(to.getField().getType())) {
```

Comment:
we should check the validity bit for list vector too. per the [spec example](https://github.com/apache/arrow/blob/master/docs/source/format/Layout.rst#example-layout-listchar-array), an empty list is different from null list.

Reply:
Thanks, fixed.